src/plugins/auto/executable_network.cpp

```diff
@@ -669,30 +670,55 @@ InferenceEngine::Parameter MultiDeviceExecutableNetwork::GetMetric(const std::st
             } else {
                 IE_ASSERT(_loadContext[CPU].isAlready == true);
-                real = _loadContext[CPU].
-                    executableNetwork->GetMetric(name).as<unsigned int>();
                 std::unique_lock<std::mutex> lock(_confMutex);
                 auto deviceInfo =  _loadContext[ACTUALDEVICE].deviceInfo;
                 lock.unlock();
                 if (deviceInfo.deviceName.find("GPU") != std::string::npos) {
                     const auto& mode = deviceInfo.config.find(CONFIG_KEY(PERFORMANCE_HINT));
                     if (mode != deviceInfo.config.end() && mode->second == CONFIG_VALUE(THROUGHPUT)) {
-                         std::map<std::string, InferenceEngine::Parameter> options;
-                         options["MODEL_PTR"] = _network.getFunction(); // CNNntework
-                         try {
-                             auto optimalBatchSize = _core->GetMetric(deviceInfo.deviceName,
-                                     METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
-                             auto rangeOfStreams = _core->GetMetric(deviceInfo.deviceName,
-                                     METRIC_KEY(RANGE_FOR_STREAMS), options).as<std::tuple<unsigned int, unsigned int>>();
-                             real = (std::max)(real, std::get<1>(rangeOfStreams) * optimalBatchSize);
-                         } catch (const InferenceEngine::Exception &iie) {
-                             LOG_WARNING("[AUTOPLUGIN]get optimal infer requset num for GPU auto-batch failed :%s", iie.what());
-                         }
+                        // THROUGHPUT mode
+                        real = 4u;
+                        // if enable auto-batch, use below code
+                        // std::map<std::string, InferenceEngine::Parameter> options;
+                        // options["MODEL_PTR"] = _network.getFunction(); // CNNntework
+                        // try {
+                        //     auto optimalBatchSize = _core->GetMetric(deviceInfo.deviceName,
+                        //             METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
+                        //     auto rangeOfStreams = _core->GetMetric(deviceInfo.deviceName,
+                        //             METRIC_KEY(RANGE_FOR_STREAMS), options).as<std::tuple<unsigned int, unsigned int>>();
+                        //     real = (std::max)(real, std::get<1>(rangeOfStreams) * optimalBatchSize);
+                        // } catch (const InferenceEngine::Exception &iie) {
+                        //     LOG_WARNING("[AUTOPLUGIN]get optimal infer requset num for GPU auto-batch failed :%s", iie.what());
+                        // }
+                    } else {
+                        // if it is not THROUGHPUT mode, we think it's LATENCY mode
+                        real = 1u;
+                    }
+                } else if (deviceInfo.deviceName.find("VPUX") != std::string::npos) {
```

Comment:
Let's check if latency mode on VPUX and MYRIAD need additional setup for nstream and nthreads

Reply:
test it, on VPU and MYRIAD, if the num is 1 ,  latency will be better

Reply:
VPUX and MYX are also operating #streams (not sure they support the Range_For_Streams though)... the bad thing with all those if/else is that we use explicit device names too much ( whihc is not scalable/future-proof)

Reply:
This is temporary solution in 2022.1 release. 

We will seek OPTIMAL_NUMBER_OF_INFER_REQUESTS as metrics of device plugin as 2022.1 release.